kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -492,53 +454,71 @@ define([
             /**
              * @param {Object} message
              */
-            handleJobStatus(self, message) {
+            handleJobStatus(self, message, addressArgs) {
+                const updated = [];
+                Object.values(message).forEach((jobStatusMessage) => {
+                    if (!jobStatusMessage.error) {
+                        updated.push(self._handleJobStatus(jobStatusMessage, addressArgs));
```

Comment:
If the job status has been updated, it is returned by `self._handleJobStatus`; this means all the updates can be collected and performed at once, rather than having a load of individual update events.